cmd/smarty-usps-comparison/main.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/corbaltcode/usps/internal/smarty"
+)
+
+type ZipcodeResult struct {
+	Zipcode      string
+	USPSFips     []string
+	SmartyFips   []string
+	ErrorMessage string
+}
+
+func parseCSVList(dataString string) []string {
+	items := strings.Split(dataString, ",")
+	for i, item := range items {
+		items[i] = strings.TrimSpace(item)
+	}
+	return items
+}
+
+func readZipToCountyDataFromCSV(reader io.Reader) ([]string, map[string][]string, error) {
+	csvReader := csv.NewReader(reader)
+	header, err := csvReader.Read()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to read header: %w", err)
+	}
+
+	if len(header) < 2 || header[0] != "ZIP Code" || header[1] != "County Numbers" {
+		return nil, nil, fmt.Errorf("unexpected header format")
+	}
+
+	var zips []string
+	zipToCounty := make(map[string][]string)
+
+	for {
+		record, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to read record: %w", err)
+		}
+
+		zip := record[0]
+		counties := parseCSVList(record[1])
+
+		zips = append(zips, zip)
+		zipToCounty[zip] = counties
+	}
+
+	return zips, zipToCounty, nil
+}
+
+func ProcessSmartyResponse(responseBody []smarty.SmartyResponse, zipToCounties map[string][]string, zipcodes []string, startIndex int) ([]ZipcodeResult, error) {
```

Comment:
That works to avoid out-of-bounds errors. I think even in the last batch, `len(batch) == len(responseBody)`. (`len(batch)` may be less than `batchSize` due to lines 194-195.)

Reply:
Oh, right! Will change to `==`